util/auto_skip_compressor.h

```diff
@@ -32,6 +32,39 @@ class CompressionRejectionProbabilityPredictor {
   size_t window_size_;
 };
 
+template <typename T>
+class WindowAveragePredictor {
```

Comment:
As a follow up/low priority separate PR, it seems like CompressionRejectionProbabilityPredictor can use this class too?